app.py

```diff
@@ -154,6 +154,15 @@ def api_search():
     return flask.jsonify(output)
 
 
+@app.route("/workout", methods=["GET", "POST"])
```

Comment:
The `methods` argument to this decorator is unnecessary; the workout page should not accept a POST request.